common/types/proposal.go

```diff
@@ -45,6 +45,11 @@ type InnerProposal struct {
 	Ballot
 	// smesher's content proposal for a layer
 	TxIDs []TransactionID
+	// aggregated hash up to the layer before this proposal.
+	MeshHash Hash32
+	// state root up to the layer before this proposal.
+	// note: this is needed in addition to mesh hash to detect bug in SVM
+	StateHash Hash32
```

Comment:
i thought that we agreed to drop this field for genesis. we will use it internally in the observability tooling to detect such bugs

Reply:
my recollection is that we use some kind of hash (the account diff) for this purpose for now. do you see a big problem here? i am ok w dropping it but thought once we have a real account state, we dont have to change node code. @dshulyak

Reply:
i personally don't see any issues if we will keep it. besides that after commitment trie will be introduced this hash will be computed differently and it may cause some confusion if  commitment trie will be built starting from older layers.

Reply:
it may force us to do a hard fork when account state is ready, but i dont see how that (account state) isnt a hard fork anyway. but i can leave it out out Proposal for now and keep the warning log.

Reply:
commitments and account state is not the same thing

Reply:
i had assume that this is the root of state trie. what is commitment then?

Reply:
i was referring to the comment about account state. account state is a leaf in the state trie, but it doesn't have to be coupled with the state trie. so e.g. when account state is read or updated there is no reason to even think about state trie.

state trie is used as a state commitment mechanism, and can be implemented relatively independently.